lib/dpl/providers/pages/api.rb

```diff
@@ -0,0 +1,102 @@
+require 'timeout'
+
+module Dpl
+  module Providers
+    class Pages
+      class Api < Pages
+        PAGES_PREVIEW_MEDIA_TYPE = 'application/vnd.github.mister-fantastic-preview+json'
+
+        # suppress warnings about preview API
+        ENV['OCTOKIT_SILENT'] = 'true'
+
+        TIMEOUTS = {
+          timeout: 180,
+          open_timeout: 180
+        }
+
+        gem 'octokit', '~> 4.14.0'
+
+        status :alpha
+
+        full_name 'GitHub Pages (API)'
+
+        description sq(<<-str)
+          This provider requests GitHub Pages build for the repository given by
+          the `--repo` flag, or the current one, if the flag is not given.
+          Note that `dpl` does not perform any check about the fitness of the request;
+          it is assumed that the target repository (and the branch that GitHub Pages is
+          configured to use) is ready for building.
+          For example, if your GitHub Pages is configured to use `gh-pages` but the
+          deployment is run on the `master` branch, you would have to ensure that the
+          `gh-pages` would be updated accordingly during the build.
+        str
+
+        opt '--repo SLUG', 'GitHub repo slug', default: :repo_slug
+        opt '--github_token TOKEN', 'GitHub oauth token with repo permission', required: true, secret: true
+
+        msgs pages_not_found:             'GitHub Pages not found for %{slug}. ' \
+               'Given token has insufficient scope (\'repo\' or \'public_repo\'), ' \
+               'or GitHub Pages is not enabled for this repo (see https://github.com/%{slug}/settings)',
+             build_pages_request_timeout: 'GitHub Pages build request timed out',
+             deploy_start:                'Requesting GitHub Pages build using API'
+
+        def validate
+          error :pages_not_found unless pages_enabled?
+        end
+
+        def deploy
+          info :deploy_start
+
+          api.request_page_build slug
```

Comment:
Does this just deploy whatever's in the repo already w/o actually pushing the new updated content?